tests/console/salt.pm

```diff
@@ -61,6 +61,7 @@ EOF
     unless (script_run 'salt \'*\' test.ping') {
         assert_script_run 'for i in {1..7}; do echo "try $i" && salt \'*\' test.ping -t30 && break; done';
     }
+    assert_screen 'True'
```

Comment:
Now this will fail because there is no needle with the tag "True" and honestly there should not be one. There should be no need to do a screen based check for the exit code analysis.

Line L61 has been introduced in 73353df6d by me and I do not even understand how that was supposed to work correctly :D

So could it be this never worked and was always the wrong way around? `script_run` returns the exit code of the command which is probably non-zero, evaluating as true in perl context so that line L62 is skipped. We have three scenarios to cover:

1. test.ping works on first call -> should pass
2. test.ping fails on first call, succeeds in retry -> should pass
3. test.ping fails on all calls with retry -> should fail

Reply:
Okay, I had no clue about needles and just gave it a shot.

But I can tell you what's going on here and why that loop was added. When master and minion are started, the minion is not yet authorized by the master. This is done with the `salt-key --accept-all -y`. But the minion is not immediately after the acceptance connecting to the master, since the minion is in a wait loop with increasing time intervals. So this can take 5 or 40 seconds. We don't know. That's why the loop was added in the test.
But now we have the problem that the exit code was ignored. That's why I had the idea to check the text return we geht there. But perhaps this is a side effect of running that in a loop?

Your outlined scenarios are spot on!

If you have the bandwidth to care of that, I'd very much appreciate that!

Reply:
sure, handled in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10229 . I will close your PR instead if you don't mind.

Reply:
Of course I don't mind. Thanks a lot!